src/pim_apps/pim_api.py

```diff
@@ -414,111 +414,136 @@ def process_pim_product(self, product, process_product):
-            df_solo = df_solo.where(pd.notnull(df_solo), None)
+    
+            # Remove Nan
+            df_parent = df_parent.where(pd.notnull(df_parent), None)
             df_variant = df_variant.where(pd.notnull(df_variant), None)
-            final_list = df_variant.to_dict("records") + df_solo.to_dict("records")
-            return final_list
-
-        if include_variants and self.pim_channel_api.group_by_parent:
-            for col in columns_to_drop:
-                    if col in df.columns:
-                        df = df.drop(col, axis=1)
-            df = df.where(pd.notnull(df), None)
-            final_list = df.to_dict('records')
-            return final_list
-
-        
+            df_solo = df_solo.where(pd.notnull(df_solo), None)
+    
+            
+            # If include_variants are FALSE, group_by_parent is TRUE and  exclude_pim_properties is TRUE
+            # then, the variant product inside the parent product wouls still have the pim properties
+            # please handle this inside the job code
+            # in this case, PRENT products will have their VARIANTS inside them in "variants" column. This will also include SOLO products
+            if not df_variant.empty:
+                grouped_variants = df_variant.groupby('pimParentId').apply(lambda x: x.to_dict('records')).reset_index()
+                grouped_variants.columns = ['pimUniqueId', 'variants']
         
-        # Separate parent, variant, and solo products
-        df_parent = df[df['pimProductType'] == 'PARENT']
-        df_variant = df[df['pimProductType'] == 'VARIANT']
-        df_solo = df[df['pimProductType'] == 'SOLO']
-
-        # Remove Nan
-        df_parent = df_parent.where(pd.notnull(df_parent), None)
-        df_variant = df_variant.where(pd.notnull(df_variant), None)
-        df_solo = df_solo.where(pd.notnull(df_solo), None)
-
+                merged_df = pd.merge(df_parent, grouped_variants, on='pimUniqueId', how='left')
         
-
-        grouped_variants = df_variant.groupby('pimParentId').apply(lambda x: x.to_dict('records')).reset_index()
-        grouped_variants.columns = ['pimUniqueId', 'variants']
-
-        merged_df = pd.merge(df_parent, grouped_variants, on='pimUniqueId', how='left')
-
-        merged_df['variants'] = merged_df['variants'].apply(lambda x: x if isinstance(x, list) else [])
-
-        if exclude_pim_properties:
-            for col in columns_to_drop:
-                    if col in merged_df.columns:
-                        merged_df = merged_df.drop(col, axis=1)
-
-        if exclude_pim_properties:
-            for col in columns_to_drop:
-                    if col in df_solo.columns:
-                        df_solo = df_solo.drop(col, axis=1)
-                        
+                merged_df['variants'] = merged_df['variants'].apply(lambda x: x if isinstance(x, list) else [])
+            else:
+                merged_df = pd.DataFrame()
```

Comment:
This should be df_solo right?

Reply:
df_solo is added later seperately.
merged_df should have parent along with variants